### PR TITLE
RFC for exporting images in parallel

### PR DESCRIPTION
**Pull request by ESWZY**
We already have some discussion here: [lifecycle#1167](https://github.com/buildpacks/lifecycle/pull/1167).

[Readable](https://github.com/ESWZY/buildpacks-rfcs/blob/main/text/0000-lifecycle-parallel-export.md)

### CODE REVIEW

**Comment by jjbustamante** [comment]
Hi @ESWZY!

Thank you so much for working on this proposal and the RFC it is really awesome!

My suggestions:

- Add to the *what is this* section the propose flag to the [exporter](https://github.com/buildpacks/spec/blob/main/platform.md#exporter) that will enable the feature.  Usually we just add a table, similar to the one in the spec, with the new flag we want to incorporate.
- Once you have the flag, in the *how it works* section, you can just add a few examples of invocations of the exporter with the new flag and you can explain the expected outcome and behavior. 

The new flags, at the end, will require a new API spec, so you can also put them under the *Spec. Changes * section, but that is something can be done later once you get consensus from the community around the feature.

**Comment by ESWZY** [comment]
> Hi @ESWZY!
> 
> Thank you so much for working on this proposal and the RFC it is really awesome!
> 
> My suggestions:
> 
> * Add to the **what is this** section the propose flag to the [exporter](https://github.com/buildpacks/spec/blob/main/platform.md#exporter) that will enable the feature.  Usually we just add a table, similar to the one in the spec, with the new flag we want to incorporate.
> * Once you have the flag, in the **how it works** section, you can just add a few examples of invocations of the exporter with the new flag and you can explain the expected outcome and behavior.
> 
> The new flags, at the end, will require a new API spec, so you can also put them under the **Spec. Changes** section, but that is something can be done later once you get consensus from the community around the feature.

Thanks for your suggestion!

I added a proposal flag to this PRC and made a preliminary attempt in [lifecycle#1167](https://github.com/buildpacks/lifecycle/pull/1167).

File: text/0000-lifecycle-parallel-export.md
Comment (jjbustamante) [inline text/0000-lifecycle-parallel-export.md:118]: I would suggest adding a `WARN` message to the end-user to indicate that using the `parallel` flag without exporting a cache image has no effect.

File: text/0000-lifecycle-parallel-export.md
Comment (jjbustamante) [inline text/0000-lifecycle-parallel-export.md:150]: From my point of view:
- How to allow users to choose the export method? environment variable? Or a parameter of the creator? 
I like how you are proposing it, with the flag the env variable

- Does it also need to be specified in the pack tool?
Probably, once we get the RFC approved, we can create the issue on `pack` to expose the behavior for end-users

- Should this feature be enabled by default?
I'd say **no**  to avoid unexpected behaviors to people consuming the *lifecycle*.

**Comment by jjbustamante** [comment]
Awesome work @ESWZY! I really like this RFC

File: text/0000-lifecycle-parallel-export.md
Comment (ESWZY) [reply text/0000-lifecycle-parallel-export.md:118]: This makes sense, I agree with you.

**Review by natalieparellano — APPROVED** [review]

**Review by jabrown85 — APPROVED** [review]

**Review by jjbustamante — APPROVED** [review]

File: text/0000-lifecycle-parallel-export.md
Comment (natalieparellano) [inline text/0000-lifecycle-parallel-export.md:72]: Suggestion from working group - when these flags are combined in the `creator` `-parallel-export` might make more sense. However `-parallel-export` might look funny as a flag to the exporter.

Perhaps `-parallel-export` as a flag to the creator and `-parallel` as a flag to the exporter? I don't have a strong opinion either way, but thought I'd surface this idea.

Comment (natalieparellano) [reply text/0000-lifecycle-parallel-export.md:72]: @ESWZY @jabrown85 perhaps you could opine here? Then I'll move this one back into voting. We already have enough approvals.

**Review by AidanDelaney — APPROVED** [review]

**Comment by natalieparellano** [comment]
Moved to voting, as this is a sub-team RFC and has approvals from both implementation maintainers I think we can merge it early cc @jabrown85

**Pull request merged**
